### PR TITLE
Refactor email tests

### PR DESCRIPTION
**Pull request by opqdonut**
follow up to #1612, for #1366 

# Definition of Done / Review checklist

## Reviewability
- [x] link to issue

## Follow-up
- [x] no critical TODOs left to implement

### CODE REVIEW

File: test/clj/rems/test_locales.clj [hunk 1]
```diff
@@ -26,6 +26,29 @@
...
+                (is (string? right))
+                (is (= (extract-format-parameters left)
+                       (extract-format-parameters right))))
+              (map? left)
+              (doseq [k (keys left)]
+                (is (map? right))
+                (testing k
+                  (check (get left k) (get right k))))))]
+    (testing "[:en vs :fi]"
+      (check (loc-en) (loc-fi)))))
```
Comment (okahilak) [inline test/clj/rems/test_locales.clj]: Nice check

File: test/clj/rems/poller/test_email.clj [hunk 2]
```diff
@@ -81,22 +78,39 @@
...
-                  user-settings/get-user-settings (constantly {:language :en})]
-      (mapv #(sort-emails (#'rems.poller.email/event-to-emails-impl % application)) events))))
+(defn email-recipient [email]
+  (or (:to email) (:to-user email)))
+
+(defn sort-emails [emails]
+  (sort-by email-recipient emails))
+
+(defn email-recipients [emails]
+  (set (mapv #(or (:to %) (:to-user %)) emails)))
```
Comment (okahilak) [inline test/clj/rems/poller/test_email.clj]: minor: could use email-recipient function here

File: test/clj/rems/poller/test_email.clj [hunk 3]
```diff
@@ -81,22 +78,39 @@
...
+
+(defn email-to [user emails]
+  ;; return arbitrary email if none match to get better errors from tests
+  (or (first (filter #(= user (email-recipient %)) emails))
+      (first emails)))
 
-(deftest test-event-to-emails-impl
-  (let [base-events [{:application/id 7
+(defn emails
+  ([lang base-events event]
```
Comment (okahilak) [inline test/clj/rems/poller/test_email.clj]: minor: if lang was the last parameter, the order of parameters would be retained between different arities.

File: test/clj/rems/poller/test_email.clj (hunk 2 above)
Comment (opqdonut) [reply test/clj/rems/poller/test_email.clj]: oh right that's what I meant to do

File: test/clj/rems/poller/test_email.clj (hunk 3 above)
Comment (opqdonut) [reply test/clj/rems/poller/test_email.clj]: I prefer the aesthetics of lang being first for some reason

Comment (okahilak) [reply test/clj/rems/poller/test_email.clj]: Ok, I guess it doesn't matter much.

**Pull request merged**